src/app/project/project-documents/upload/upload.component.ts

```diff
@@ -150,6 +170,12 @@ export class UploadComponent implements OnInit, OnDestroy {
       );
   }
 
+  /**
+   * Make sure the document name doesn't include any invalid characters
+   * (that wouldn't work within a URL, for example).
```

Comment:
```suggestion
   * that wouldn't work within a URL.
```
Not sure if you have a location for re-usable function, this would probably fall in that category. maybe for another time.